server/stt/stt_events_ml.py

```diff
@@ -1,12 +1,45 @@
+from typing import Optional, Dict, Any
+
+import logging
 from xml.etree.ElementTree import Element
+from bson import ObjectId
 
+from superdesk import get_resource_service
 from superdesk.utc import local_to_utc
 from superdesk.io.registry import register_feed_parser
 from superdesk.text_utils import plain_text_to_html
+from superdesk.errors import SuperdeskApiError
 from planning.feed_parsers.events_ml import EventsMLParser
 
+logger = logging.getLogger(__name__)
 TIMEZONE = "Europe/Helsinki"
 
+NS = {
+    "stt": "http://www.stt-lehtikuva.fi/NewsML",
+}
+
+
+def search_existing_contacts(contact: Dict[str, Any]) -> Optional[Dict[str, Any]]:
+    """Attempt to find existing media contact using email, falling back to first_name/last_name combo"""
+
+    contacts_service = get_resource_service("contacts")
+    if len(contact.get("contact_email") or []):
+        cursor = contacts_service.search(
+            {"query": {"bool": {"must": [{"term": {"contact_email.keyword": contact["contact_email"][0]}}]}}}
```

Comment:
it might be useful to call `.lower()` on the email as it won't match if we get the email from ingest with some uppercase